### ISSUE TITLE
https basepath is not handled in SwaggerSpecConverter

### ISSUE THREAD

**Issue by fragag03** [swagger-api/swagger-js#398]
A basepath that start with https is not set

found the responsible code (line 1320 in version 2.1.2-M2):

```javascript
  var basePath = obj.basePath;
  if(obj.basePath.indexOf('http://') === 0) {
    var p = obj.basePath.substring('http://'.length);
    var pos = p.indexOf('/');
    if(pos > 0) {
      swagger.host = p.substring(0, pos);
      swagger.basePath = p.substring(pos);
    }
    else{
      swagger.host = p;
      swagger.basePath = '/';
    }
  }
```
As you see, it only handle http url and not https...

I changed it to this and it seems to make it works:

```javascript
var basePath = obj.basePath;
  var basePathProtocol;

  if(obj.basePath.indexOf('http://') === 0)
      basePathProtocol = 'http://';
  else if (obj.basePath.indexOf('https://') === 0)
      basePathProtocol = 'https://';

  if (basePathProtocol){
    var p = obj.basePath.substring(basePathProtocol.length);
    var pos = p.indexOf('/');
    if(pos > 0) {
      swagger.host = p.substring(0, pos);
      swagger.basePath = p.substring(pos);
    }
    else{
      swagger.host = p;
      swagger.basePath = '/';
    }
  }
```

**Comment by webron** [comment]
That looks like just one tiny step before submitting a PR with the change against the develop_2.0 branch ;)

**Comment by fragag03** [comment]
Ok done. It's my first time doing this so I hope everything is ok.

**Comment by fehguy** [comment]
fixed in #399
